### ISSUE TITLE
Documentation: Clarification on Integer operators

### ISSUE THREAD

**Issue by devilHorns13** [golang/go#52398]
On the Go documentation around [Integer operators](https://go.dev/ref/spec#Integer_operators), it's quoted:

`
If the dividend is non-negative and the divisor is a constant power of 2, the division may be replaced by a right shift
`

Though the accompanying example shows it clearly, it's stated as if it's only the division that gets replaced by a right shift, where in fact you have to replace the right-hand operand, `x = 2^n` with `n`.

**Comment by robpike** [comment]
As I believe that whole section makes clear, as there are other examples, it's the overall operation that may be implemented another way. It would be incorrect just to replace the operator with a shift without adjusting the operands; surely that doesn't need explanation?

If anything, that section could probably just remove the text describing possible implementations, as it's irrelevant to the spec and the compiler does many similar optimizations without them needing to be called out.

**Comment by beoran** [comment]
@robpike I agree that it is a good idea to omit discussion on specific optimizations, but it may be good to replace this by a mention that the go compiler may optimize arithmetic by rewriting the expression as long as the result remains the same.
